### ISSUE TITLE
[Feature Request] "Wiping Battery" Option

### ISSUE THREAD

**Issue by E-Gwen** [AsteroidOS/asteroid#229]
Since I flashed my Asus Zen Watch 2 "Sparrow", the battery behaves as a weird one.

So i was asking if you could add a "Wipe Battery" option in AsteroidOS ?

It would be helpful, i suppose, for the entire community.

Won't you think about it, will you ?

Best regards.

Happy festivities to all.

**Comment by beroset** [comment]
I'm not sure I understand.  What would "wipe battery" do exactly?

**Comment by parkerlreed** [comment]
I think they mean calibration. 

If it's anything like my Zenwatch 2, the battery is a lost cause anyways without replacement.

**Comment by E-Gwen** [comment]
Yes, to calibrate the battery so that the battery percentage information keeps meanfull.

@parkerlreed : 
Don't you know you could save several items by making a efficient wipe battery, on purpose ?
It's my case, that's why i'm asking for this feature option.
Maybe not for the ASUS Zen watch 2, that will be a pity.

Thanks for taking care.

AsteroidOSly yours.  ;-)

**Comment by dodoradio** [comment]
@E-Gwen I don't believe there is anything we can actually do to recalibrate the battery. if it's not reaching 100% as expected when charging, something is very wrong, and it's likely that the battery has already failed significantly enough that an adjusted battery gauge won't make the watch any more useful (generally, with this kind of failure mode, you'll also see the battery running down in less than an hour)

**Comment by E-Gwen** [comment]
@dodoradio : 

"if it's not reaching 100% as expected when charging, something is very wrong, and it's likely that the battery has already failed significantly enough"

Absolutely not, it was the case for my smartwatch, and i managed to give my battery a second life.
Why do you think i'm asking for this feature ?

**Comment by beroset** [comment]
For me, it's still not clear what the feature actually *is* so I'm afraid I'm not likely to be able to help with the implementation. Battery management in wearable devices, including watches, is typically the province of the Power Management IC (PMIC) and associated driver, neither of which are under the control of AsteroidOS.  The PMIC is chosen by the Original Equipment Manufacturer (OEM) and the driver is most often a closed-source blob that we access via `libhybris`.  So any implementation is likely to require many more technical details or references than we have so far.

**Comment by E-Gwen** [comment]
All right.

If it not possible to make the battery great again, that's the way it is.

I discovered this option when i flashed an android device with an alternative rom, long years ago.
It was included in the recovery menu.

I was exciting by this option, and believed it would have been easily implemented. :-(

Good Spring.

**Comment by dodoradio** [comment]
if it's a feature of the recovery menu, then it's up to the specific device's firmware, and that's completely out of our control. Given that this is infeasible, could we close this issue?
